### DISCUSSION TITLE
Netbox 3.3.5 database performance issue

### DISCUSSION THREAD

**Discussion by aakozlov85** [netbox-community/netbox#11642] [Q&A] [answered]
Hello guys!
We 've got netbox instance v3.3.5 at VM and there are some performance issues. The database contains about 1257 devices and it take for a long time to display netbox info when i press "Devices" button (about 4 second). The database and redis server - all at the same VM.
Please have a look for browser get request time response.
![netbox_devices_responseTime](https://user-images.githubusercontent.com/94970464/216321486-bd76ba2c-3834-4b8a-8c1a-6486e4bc180e.jpg)
All other buttons (Device Type, Manufactors and etc.) works fine and it takes about 0,5 seconds to go.The problem is only with devices, as they have a lot of elements.
![netbox_device_role_responseTime](https://user-images.githubusercontent.com/94970464/216321438-b08944a3-3bf6-48cc-b016-f84509121ec3.jpg)
Please advice how to diagnose and fix it?

**Comment by kkthxbye-code** [comment]
What is your pagination setting? Please also list what columns you have selected. A likely reason is that you have a custom link column selected.

**Reply by aakozlov85** [reply]
as i know, we have default settings for pagination.
`'DEFAULT_PAGINATION_CLASS': 'netbox.api.pagination.OptionalLimitOffsetPagination',
`
Caching is disabled also (CACHE_TIMEOUT set to 0)
I do not select any columns, i am only  press "Devices" button at the netbox main page.
![netbox_devices_buttonjpg](https://user-images.githubusercontent.com/94970464/216329787-b86dc1e5-8eb4-4ed5-a711-9340865bc552.jpg)

**Reply by aakozlov85** [reply]
![image](https://user-images.githubusercontent.com/94970464/216330283-3fd0faf7-a520-41f6-a9d9-7272ec1d99dc.png)
Just checked it. It shows me 1000 items per page.

**Reply by kkthxbye-code** [reply]
Please click the configure table button and post a screenshot of all columns in the Selected Columns box.

**Reply by kkthxbye-code** [reply]
Please do note that having 1000 as the pagination size is bound to be slow for tables with a lot of rows. It's fast for other pages probably because you don't have close to 1000 objects of that type. 4 seconds seems about right for 1000 pagination size for devices, you might get it down a little by deselecting the columns you don't use, but really you should just lower the pagination size to something more reasonable, like 50 or 100.

**Reply by aakozlov85** [reply]
![image](https://user-images.githubusercontent.com/94970464/216332272-b125bda1-4479-46c7-9e20-6fc01cf5c86f.png)

**Reply by aakozlov85** [reply]
> but really you should just lower the pagination size to something more reasonable, like 50 or 100.

ok. how to accomplish this for all users? are there any settings for this?

**Reply by kkthxbye-code** [reply]
The default is 50 or 100 I believe, you have changed it yourself for your user to 1000 most likely.

There's a couple of options:

https://docs.netbox.dev/en/stable/configuration/miscellaneous/#max_page_size

Sets the maximum page size, this also applies to the API though.

https://docs.netbox.dev/en/stable/configuration/default-values/#default_user_preferences

Allows you to set the default pagination size, but as I said the default is already lower than 1000.

**Comment by aakozlov85** [comment, marked as answer]
yes you are right.
we 've MAX_PAGE_SIZE = 1000.
i decreased this value to 25 and now devices page download speed about 2.5 seconds.

What about cache settings? Can i set cache_timeout to improve performance one more?

**Reply by kkthxbye-code** [reply]
Queryset caching was removed in v3.0 so CACHE_TIMEOUT doesn't do anything anymore.

**Reply by aakozlov85** [reply]
ok. thanks a lot.
Are there any way to set selected columns for all users in table configuration?
